game/src/contrib/components/ItemComponent.java

```diff
@@ -4,46 +4,49 @@
 
 import core.Component;
 import core.Entity;
+import core.utils.TriConsumer;
 
 /**
- * A class that marks an entity as an Item and controls its {@link ItemData}
+ * Marks an entity as an Item.
  *
- * <p>It contains the {@link #itemData}, which contains all info about the Item.
+ * <p>It contains the {@link #itemData}, which contains all information about the Item.
+ *
+ * <p>An entity with an {@link ItemComponent} is not an item in an inventory, but an item that
+ * exists in the game world. Systems such as {@link contrib.systems.CollisionSystem CollisionSystem}
```

Comment:
ChatGPT? Hier fehlt imho eine Art Einleitung. Vielleicht so in der Art:

Ein Item wird im Spiel als eine Entität mit einer ItemComponent repräsentiert. Dabei speichert die ItemComonent eine Referenz auf ein ItemData-Objekt, welches sämtliche Informationen zu dem Item darstellt. 

Wenn ein Item in ein Inventar aufgenommen werden soll, wird nur das ItemData-Objekt aufgenommen - damit verliert das Item die Eigenschaften einer Entität und ist nicht mehr direkt im Spiel interagierbar. Wenn das Item aus dem Inventar wieder im Spiel abgelegt (gedroppt) wird, dann wird eine neue Entität mit einer ItemComponent um das ItemData-Objekt herum gebaut, damit das Item wieder normal interagierbar wird.

Reply:
> ChatGPT?

Ne ne, ein wenig mist kann ich auch alleine verzapfen. 

Deine Beschreibung stimmt grundsätzlich, beschreibt aber die Umsetzung eines Items. Man müsste das jetzt so formulieren, dass klar wird, dass das ItemComponent nicht dafür sorgt, dass das alles passiert, sondern man das selbst programmieren muss (bzw. dass der WorldItemGenerator das so gemacht hat).

Reply:
Genau, ich habe die Funktionsweise eines Items beschrieben. Das fehlte mir in Deinem Text als Einleitung. Ich stimme Dir aber zu, dann man dann auch noch einen Satz ergänzen sollte, dass dieses Verhalten bereits hier und dort so implementiert ist.

Reply:
habs mal drastisch überarbeitet

Reply:
ja, besser 👍